src/pconn.cc

```diff
@@ -348,36 +350,44 @@ PconnPool::key(const Comm::ConnectionPointer &destLink, const char *domain)
 }
 
 void
-PconnPool::dumpHist(StoreEntry * e) const
+PconnPool::dumpHist(std::ostream &yaml) const
 {
-    storeAppendPrintf(e,
-                      "%s persistent connection counts:\n"
-                      "\n"
-                      "\t Requests\t Connection Count\n"
-                      "\t --------\t ----------------\n",
-                      descr);
+    AtMostOnce explanation(
+        "  connection use histogram: \n"
+        "    # requests per connection: closed connections that carried that many requests\n");
 
     for (int i = 0; i < PCONN_HIST_SZ; ++i) {
         if (hist[i] == 0)
             continue;
 
-        storeAppendPrintf(e, "\t%d\t%d\n", i, hist[i]);
+        yaml << explanation <<
+             "    " << i << ": " << hist[i] << "\n";
     }
 }
 
 void
-PconnPool::dumpHash(StoreEntry *e) const
+PconnPool::dumpHash(std::ostream &yaml) const
 {
-    hash_table *hid = table;
+    const auto hid = table;
     hash_first(hid);
+    AtMostOnce title("  open connections list:\n");
 
-    int i = 0;
-    for (hash_link *walker = hash_next(hid); walker; walker = hash_next(hid)) {
-        storeAppendPrintf(e, "\t item %d:\t%s\n", i, (char *)(walker->key));
-        ++i;
+    for (auto *walker = hash_next(hid); walker; walker = hash_next(hid)) {
+        yaml << title <<
+             "    \"" << static_cast<char *>(walker->key) << "\": " <<
+             static_cast<IdleConnList *>(walker)->count() <<
+             "\n";
     }
 }
 
+void
+PconnPool::dump(std::ostream &yaml) const
+{
+    yaml << "pool " << description() << "\n";
```

Comment:
Still missing a column?

```suggestion
    yaml << "pool " << descr << ":\n";
```

The above suggestion also stops using branch-added description() because that function is currently not needed (here, is not used anywhere else) and should not be added.

Reply:
There were more missing, this must have slipped. Thanks